lib/experimental/fonts-library/class-wp-fonts-library.php

```diff
@@ -0,0 +1,618 @@
+     */
+    function get_google_fonts () {
+        $file = file_get_contents(
+            path_join ( dirname(__FILE__),"google-fonts.json" )
+        );
+        if ( $file ) {
+            return new WP_REST_Response( json_decode( $file ) );
+        }
+        return new WP_Error(
+            'rest_cant_read_google_fonts',
+            __( 'Error reading Google Fonts JSON file.' ),
+            array( 'status' => 500 )
+        );
+    }
+
+    /**
+     * Deletes a specified font asset file from the fonts directory.
+     *
+     * @param string $src The path of the font asset file to delete.
+     * @return bool Whether the file was deleted.
+     */
+    function delete_asset( $src ) {
+        $filename = basename( $src );
+        $file_path = path_join( $this->wp_fonts_dir, $filename );
+
+        if ( ! file_exists( $file_path ) ) {
+            return false;
+        }
+
+        return wp_delete_file( $file_path );
+    }
+
+    /**
+     * Deletes all font face asset files associated with a given font face.
+     *
+     * @param array $font_face The font face array containing the 'src' attribute with the file path(s) to be deleted.
+     */
+    function delete_font_face_assets ( $font_face ) {
+        if ( is_array( $font_face['src'] ) ) {
+            foreach ( $font_face['src'] as $src ) {
+                $was_assets_removed = $this->delete_asset($src);
+                if ( ! $was_assets_removed ) {
+                    // Bail if any of the assets could not be removed
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            return $this->delete_asset( $font_face['src'] );
+        }
+    }
+
+    /**
+     * Downloads a font asset.
+     *
+     * Downloads a font asset from a specified source URL and saves it to the font directory.
+     *
+     * @param string $src The source URL of the font asset to be downloaded.
+     * @param string $filename The filename to save the downloaded font asset as.
+     * @return string|bool The relative path to the downloaded font asset. False if the download failed.
+     */
+    function download_asset( $src, $filename ) {
+        $file_path = path_join( $this->wp_fonts_dir, $filename );
+
+        // Checks if the file to be downloaded has a font mime type
+        if ( ! $this->has_font_mime_type( $file_path ) ) {
+            return false;
+        }
+
+        // Downloads the font asset or returns false
+        $temp_file = download_url( $src );
+        if ( is_wp_error( $temp_file ) ) {
+            @unlink( $temp_file );
+            return false;
+        }
+
+        // Moves the file to the fonts directory or return false
+        $renamed_file = rename( $temp_file, $file_path );
+        if ( ! $renamed_file ) {
+            @unlink( $temp_file );
+            return false;
+        }
+
+        // Cleans the temp file
+        @unlink( $temp_file );
```

Comment:
It seems that `$temp_file` is being deleted regardless of the value of `$renamed_file`, so I would refactor that code.
```suggestion
        // Cleans the temp file
        @unlink( $temp_file );
        
        if ( ! $renamed_file ) {            
            return false;
        }
```

Reply:
Implemented here: https://github.com/WordPress/gutenberg/pull/50927/commits/645626f1f8b9206c268a4dbc7f3b05ea25e17ac6